src/Node/ClassPropertiesNode.php

```diff
@@ -219,6 +219,9 @@ public function getUninitializedProperties(
 				}
 			} elseif (array_key_exists($propertyName, $initializedPropertiesMap)) {
 				$hasInitialization = $initializedPropertiesMap[$propertyName]->or($usageScope->hasExpressionType(new PropertyInitializationExpr($propertyName)));
+				if ($usageScope->isInAnonymousFunction() && $usageScope->getParentScope() !== null) {
+					$hasInitialization = $hasInitialization->or($usageScope->getParentScope()->hasExpressionType(new PropertyInitializationExpr($propertyName)));
+				}
```

Comment:
maybe only enter this IF, if the outer `!hasInitialization->yes()` ?

Reply:
Good point, thanks.

I also moved the initialization of `$hasInitialization` below the IF which ignores functions other than the constructor (https://github.com/phpstan/phpstan-src/pull/2607/files#diff-0e5b4f9339ad4cf44915b25c9ea53bf150de4b6fd060856247cc74b0fc0c54eaR225). Since the conditions in that IF do not depend on `$hasInitialization`, we can `continue` the loop earlier. I hope that's ok.